sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -483,9 +513,122 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
         "programs.intlExchangeProgramEligibility",
         "programs.programDeclaration",
         "institution.id",
+        "institution.legalOperatingName",
+        "programs.submittedOn",
+        "submittedBy.firstName",
+        "submittedBy.lastName",
+        "statusUpdatedBy.firstName",
+        "statusUpdatedBy.lastName",
+        "programs.statusUpdatedOn",
+        "programs.effectiveEndDate",
       ])
+      .leftJoin("programs.submittedBy", "submittedBy")
+      .leftJoin("programs.statusUpdatedBy", "statusUpdatedBy")
       .innerJoin("programs.institution", "institution")
       .where("programs.id = :id", { id: programId })
       .getOne();
   }
+
+  /**
+   * Approve a pending program.
+   * this is a db transaction performing
+   * 3 functions, creating note, update the
+   * program and adding institution notes
+   * @param institutionId institution id.
+   * @param programId program id.
+   * @param userId user id.
+   * @param payload ApproveProgram
+   * @param returns EducationProgram
+   */
+  async approveEducationProgram(
+    institutionId: number,
+    programId: number,
+    userId: number,
+    payload: ApproveProgram,
+  ): Promise<void> {
+    return this.connection.transaction(async (transactionalEntityManager) => {
+      // create Note
+      const notes = new Note();
+      notes.description = payload.approvedNote;
+      notes.noteType = NoteType.Program;
+      notes.creator = {
+        id: userId,
+      } as User;
+      const noteObj = await transactionalEntityManager
+        .getRepository(Note)
+        .save(notes);
+
+      // update program
+      const program = new EducationProgram();
+      program.id = programId;
+      program.approvalStatus = ApprovalStatus.approved;
+      program.statusUpdatedOn = new Date();
+      program.effectiveEndDate = new Date(payload.effectiveEndDate);
+      program.statusUpdatedBy = { id: userId } as User;
+      program.programNote = noteObj;
+      // await this.repo.save(program);
+
+      await transactionalEntityManager
+        .getRepository(EducationProgram)
+        .save(program);
+
+      // update institution note
+      await transactionalEntityManager
+        .getRepository(Institution)
+        .createQueryBuilder()
+        .relation(Institution, "notes")
+        .of(institutionId)
```

Comment:
Maybe this parameter is not critical but it seems that the expectation is provide an Entity as a parameter and not the id.
Maybe something as below?
```
.of({ id: institutionId } as Institution)
```

Reply:
both are working